src/libp2p_stream_multistream.erl

```diff
@@ -46,29 +47,27 @@
 init(Kind, Opts = #{handler_fn := HandlerFun}) ->
     init(Kind, maps:remove(handler_fn, Opts#{handlers => HandlerFun()}));
 init(server, Opts = #{handlers := Handlers}) when is_list(Handlers), length(Handlers) > 0 ->
+    lager:debug("init server multistream with handlers ~p", [Handlers]),
     NegotiationTime = maps:get(negotiation_timeout, Opts, ?DEFAULT_NEGOTIATION_TIME),
-    {ok, #state{fsm_state = handshake, handlers = Handlers, negotiation_timeout = NegotiationTime},
+    {ok, #state{fsm_state = handshake, handlers = Handlers,
+                negotiation_timeout = NegotiationTime},
         [
             {packet_spec, [varint]},
             {active, once},
             {send, encode_line(<<?PROTOCOL>>)},
             {timer, negotiate_timeout, NegotiationTime}
         ]};
 init(client, Opts = #{handlers := Handlers}) when is_list(Handlers), length(Handlers) > 0 ->
+    lager:debug("init client multistream with handlers ~p", [Handlers]),
     NegotiationTime = maps:get(negotiation_timeout, Opts, ?DEFAULT_NEGOTIATION_TIME),
-    %% The client side of a multistream _should_ get a server
-    %% handshake right away. The tcp stack may combine an incoming and
-    %% an outgoing connection as part of a simultaneous
-    %% connection. This would mean we'd end up with two clients. In
-    %% order to break this deadlock we add some randomness to the
-    %% timeout and end up having one of the clients do a reverse
-    %% handshake, when successful turns the originating client into a
-    %% server.
     HandshakeTime = maps:get(handshake_timeout, Opts, rand:uniform(20000) + 15000),
-    {ok, #state{fsm_state = handshake, handlers = Handlers, negotiation_timeout = NegotiationTime},
+    {ok, #state{fsm_state = handshake, handlers = Handlers,
+                negotiation_timeout = NegotiationTime},
         [
             {packet_spec, [varint]},
             {active, once},
+            %% have client/initiator kick things off by sending the initial multistream handshake
```

Comment:
How does it kick things of when the server is _also_ sending the handshake?

Reply:
in an erlang node -> erlang node scenario the server only sends its handshake in response to a client opening a new stream and so in this case the dialler *is* the initiator.  The same is true between a rust node -> erlang node scenario as the rust implementation follows the same approach.

It may true that other libp2p implementations may differ and have the server being the initiator of the multistream handshake ( like libp2p streams was prior to this PR )

Reply:
I agree that the implementation should be the client kicking of the protocol handshake but doesn't https://github.com/helium/libp2p-streams/pull/3/files#diff-0c6dd1e33084f0989beea9aae65c1a23L54 imply the server is sending the protocol handshake too?